src/nerve_tools/configure_nerve.py

```diff
@@ -395,21 +434,23 @@ def update_subconfiguration_for_here(
         )
     for (service_name, service_info) in paasta_services:
         envoy_service_info: Optional[ServiceInfo] = None
-        if envoy_listeners:
-            key = f"{service_name}.{service_info['port']}"
-            if key in envoy_listeners:
-                envoy_service_info = copy.deepcopy(service_info)
-                envoy_ingress_port = envoy_listeners[key]
-                envoy_service_info['port'] = envoy_ingress_port
-                envoy_service_info['healthcheck_port'] = envoy_ingress_port
-                healthcheck_headers = envoy_service_info.get('extra_healthcheck_headers', {})
-                healthcheck_headers['Host'] = service_name
-                envoy_service_info['extra_healthcheck_headers'] = healthcheck_headers
+        envoy_key = f"{service_name}.{service_info['port']}"
+        if envoy_listeners and envoy_key in envoy_listeners:
+            service_info_copy = copy.deepcopy(service_info)
```

Comment:
kinda confused here. you call this `service_info_copy`, but then on line 453 you are using `envoy_service_info` not `service_info_copy`

Reply:
mypy's type determination and hinting makes this a bit complicated. 

I want envoy_service_info to be either None or ServiceInfo, so I initialize it first as None (but type hinted with Optional[ServiceInfo]. mypy will complain if I set envoy_service_info to a copy of service_info because service_info is of type SubSubConfiguration.

The simplest solution I found is to make a temporary variable called service_info_copy, modify it as needed, then cast it as ServiceInfo. Then when i make the call to update_subconfiguration, I cast it as an Optional[ServiceInfo] which satisfies both scenarios.

I just realized from your comment below that I can cast envoy_service_info in the if clause as an Optional[ServiceInfo]. This way I don't have to have a second cast when making the call to update_subconfiguration.

Reply:
From my reading, `service_info` is actually a [ServiceNamespaceConfig](https://github.com/Yelp/paasta/blob/ea308f1f801fd0835909947f2c0b83a7ab71ce47//paasta_tools/long_running_service_tools.py#L58), which doesn't define any of these attributes explicitly. It might be worthwhile to either fix paasta to give us something a bit more explicitly specified. Or stuff it into something here as soon as we get it. By that I mean wrapping the paasta function and having that return something that has explicit fields. Alternatively you could have a function that's like `service_info_from_paasta_service_namespace_config`.

Either way, it feels like we're kind of hacking the logic to work with a data model that doesn't seem to really be accurate anymore.

WDYT?